examples/apo.rs

```diff
@@ -0,0 +1,348 @@
+        children: &[miniscript::expression::Tree<'_>],
+    ) -> Result<Self, ()> {
+        if children.len() == 7 && name == "all_apo" {
+            if children.iter().any(|x| !x.args.is_empty()) {
+                return Err(());
+            }
+            let outputs = deser_hex::<Vec<TxOut>>(children[0].name)?;
+            let in_asset = deser_hex::<elements::confidential::Asset>(children[1].name)?;
+            let in_value = deser_hex::<elements::confidential::Value>(children[2].name)?;
+            let in_spk = deser_hex::<elements::Script>(children[3].name)?;
+            let version = expression::parse_num(children[4].name).map_err(|_e| ())?;
+            let locktime = expression::parse_num(children[5].name).map_err(|_e| ())?;
+            let sequence = expression::parse_num(children[6].name).map_err(|_e| ())?;
+            Ok(SighashAllAPO {
+                outputs,
+                in_asset,
+                in_value,
+                in_spk,
+                version,
+                locktime,
+                sequence,
+            })
+        } else {
+            // Correct error handling while parsing fromtree
+            Err(())
+        }
+    }
+}
+
+impl ParseableExt for SighashAllAPO {
+    fn from_token_iter(
+        _tokens: &mut miniscript::miniscript::lex::TokenIter<'_>,
+    ) -> Result<Self, ()> {
+        // Parsing back from script is currently not implemented
+        Err(())
+    }
+
+    fn evaluate<'intp, 'txin>(
+        &'intp self,
+        _stack: &mut miniscript::interpreter::Stack<'txin>,
+        txenv: Option<&miniscript::TxEnv>,
+    ) -> Result<bool, miniscript::interpreter::Error> {
+        let env = txenv.ok_or(miniscript::interpreter::Error::IncorrectCovenantWitness)?;
+        self.eval(env)
+    }
+
+    #[rustfmt::skip]
+    fn push_to_builder(&self, builder: elements::script::Builder) -> elements::script::Builder {
+        let mut builder = builder;
+        for (i, out) in self.outputs.iter().enumerate() {
+            let asset_eq = CovOps::<CovExtArgs>::AssetEq(
+                AssetExpr::Const(out.asset.into()),
+                AssetExpr::Output(i),
+            );
+            let value_eq = CovOps::<CovExtArgs>::ValueEq(
+                ValueExpr::Const(out.value.into()),
+                ValueExpr::Output(i),
+            );
+            let spk_eq = CovOps::<CovExtArgs>::SpkEq(
+                SpkExpr::Const(Spk(out.script_pubkey.clone()).into()),
+                SpkExpr::Output(i),
+            );
+            builder = asset_eq.push_to_builder(builder).push_opcode(opcodes::all::OP_VERIFY);
+            builder = value_eq.push_to_builder(builder).push_opcode(opcodes::all::OP_VERIFY);
+            builder = spk_eq.push_to_builder(builder).push_opcode(opcodes::all::OP_VERIFY);
+        }
+
+        use opcodes::all::*;
+        builder = builder
+            .push_opcode(OP_INSPECTVERSION).push_slice(&self.version.to_le_bytes()).push_opcode(OP_EQUALVERIFY)
```

Comment:
In 66aeb41968c171817a93fc30ae00471f6a8dbe90:

I would really prefer that all the `.push_` methods got their own line.

Reply:
Ok, on closer reading I see that the reason is that you've got `INSEPECTXXX <XXX> EQV` all on one line ... so I feel less strongly. But I'd still suggest breaking it up across separate lines and maybe adding comments in between to separate the sections.